### PR TITLE
Adding Sum<> Function

### PR DESCRIPTION
**Pull request by fdarosa2663**

### CODE REVIEW

File: functions/sum.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  }
+  int getInteger(int led) {
+    sum_ = a_ + b_;
+    return sum_.getInteger(led);
+  }
+
+private:
+  A a_;
+  B b_;
+  int sum_;
```
Comment (profezzorn) [inline functions/sum.h:26]: remove sum_

File: functions/sum.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+template<class A, class B>
+class Sum {
+public:
+  void run(BladeBase* blade) {
+    a_.run(blade);
+    b_.run(blade);
+  }
+  int getInteger(int led) {
+    sum_ = a_ + b_;
```
Comment (profezzorn) [inline functions/sum.h:19]: remove this line (does this even compile?)

File: functions/sum.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    return sum_.getInteger(led);
```
Comment (profezzorn) [inline functions/sum.h:20]: replace with: return a_.getInteger(led) + b_.getInteger(led);

File: functions/sum.h [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+#define FUNCTIONS_SUM_H
+
+// Usage: SUM<A, B>
+// Adds A + B
+// A, B: INTEGER
+// return value: INTEGER
+
+class BladeBase;
+
+template<class A, class B>
```
Comment (profezzorn) [inline functions/sum.h:11]: We should fix this template to be able to sum multiple things.
See LayerFunctions<> for now to do that.

File: functions/sum.h (hunk 3 above)
Comment (fdarosa2663) [reply functions/sum.h]: Thanks, misunderstood original change, updated and compiled.

File: functions/sum.h (hunk 4 above)
Comment (fdarosa2663) [reply functions/sum.h]: I modified using same approach as LayerFunctions to take multiple functions

**Comment by profezzorn** [comment]
This is basically ready to commit, but there is a conflict in ProffieOS.ino that needs to be resolved first.

**Comment by profezzorn** [comment]
Still says there is a conflict.

**Comment by profezzorn** [comment]
Ops, wrong button, didn't mean to close it.
Uh, I can't seem to reopen it....
Says the repo this came form was deleted?

**Comment by profezzorn** [comment]
Might need to create a new PR, sorry.

**Pull request closed without merging**